### ISSUE TITLE
Faulty check for /var/run/docker.sock in studio on Mac OSX

### ISSUE THREAD

**Issue by blixtra** [habitat-sh/habitat#4629]
If `/var/run/docker.sock` is not present on the OSX host, the studio, which runs in a Linux VM, is not correctly configured to use Docker.

**Background**
When using Docker on a Mac the docker-daemon runs in a Linux VM. Meaning all the docker containers also run in that VM. On Mac, when the studio is started, the studio also runs inside a container on that VM.

[This link](https://forums.docker.com/t/how-is-var-run-docker-sock-created-on-mac-if-docker-daemon-runs-in-hyperkit-vm-on-linux/29090/2) explains an OSX Docker setup a bit.

**Issue**
While the studio gets successfully deployed to a container in the guest Docker VM, the configuration logic to setup Docker for that container is faulty. This breaks the Docker export and relate functionality.

In short, the studio checks for the `/var/run/docker.sock` on OSX, but mounts it on the Docker VM.

**The hack**
To "fix" the issue, I simply did a `touch /var/run/docker.sock` on my host OSX machine. With that, the test of its existence passed (even though that file is not even a socket file) and the mount which takes place on the VM works fine.

Obviously, this should do a different type of check in this scenario.

**Comment by fnichol** [comment]
Great point and a use case that I would have considered in the pre-Docker Toolkit days.

Would it be worth checking first for `DOCKER_HOST` environment variable to catch anyone using this way to target their Docker Engine before falling back (assuming we also check for file existence too)?

**Comment by eeyun** [comment]
@blixtra, @fnichol and I started digging into this today and we had a couple of questions. Initially I thought this was definitely going to be a bug with our studio code but on further review we're not so certain.

The first and biggest question is: In the use-case you've shared here are you using the DOCKER_HOST envvar to point your docker binary to a different docker machine than what you're using locally? In our discussion we had a hard time determining in what situation you would be entering the studio on a macOS in which the docker socket wouldn't exist on your filesystem outside of that one usecase.

If that _is_ what's going on, we think this might be an issue of documentation around how to use the `HAB_DOCKER_OPTS` envvar to get around this. If that's not the situation, then having a little more context would be really useful. What we do today is we check for the docker socket and if it exists it gets volume mounted into the studio when you enter - https://github.com/habitat-sh/habitat/blob/2f22df76da8cc78b33beb72b3bbaeb8dd0a81acb/components/hab/src/command/studio/docker.rs#L75 .
